### DISCUSSION TITLE
Yaegi interpreter unexpected behavior

### DISCUSSION THREAD

**Discussion by talevy-runai** [traefik/yaegi#1548] [Q&A]
Hi,
I've written plugin for jwt validation,

I've used [github.com/golang-jwt/jwt/v4](http://github.com/golang-jwt/jwt/v4)

When I'm running Parse token, I get an empty map (only in yaegi test)
go test works fine,

I tried to debug this and I've found out the issue in the line 160

![image](https://user-images.githubusercontent.com/104714466/235474522-cfde7311-af03-4765-a951-2c0c1b7916ae.png)

the decode from object return empty string, example object:

![image](https://user-images.githubusercontent.com/104714466/235474588-79da62bc-0cc5-4799-947d-9e21a795d33d.png)

I think it doesn't work with the json:"email,omitempty" annotation.

any idea how to manage it?

 I'm using yaegi latest

**Comment by talevy-runai** [comment]
@taliesins

I think you had the same issue, am i right?

**Comment by mwitteveen** [comment]
Im experiencing the same here

```go
	parts := strings.Split(bearerToken, " ")
	if len(parts) != 2 || parts[0] != "Bearer" {
		return Claims{}, errors.New("expected authorization header format: Bearer <token>")
	}

	var claims Claims
	token, _, err := a.parser.ParseUnverified(parts[1], &claims)
	if err != nil {
		return Claims{}, fmt.Errorf("error parsing token: %w", err)
	}
	os.Stdout.WriteString(fmt.Sprintf("Token string: %s\n", parts[1]))
	os.Stdout.WriteString(fmt.Sprintf("parsed claims [%v]", claims))
	os.Stdout.WriteString(fmt.Sprintf("parsed token claims [%v]", token.Claims))
	os.Stdout.WriteString(fmt.Sprintf("parsed token [%v]", token))

	payloadBase64 := strings.Split(parts[1], ".")[1]
	payloadJSON, err := base64.RawURLEncoding.DecodeString(payloadBase64)
	if err != nil {
		fmt.Printf("Error decoding payload: %v\n", err)
	} else {
		fmt.Printf("Decoded payload: %s\n", string(payloadJSON))
	}
```

when manually decoding the payload is there otherwise the claims are empty
